components/lsp-form/MintForm.tsx

```diff
@@ -9,19 +9,103 @@ import {
 } from "./LSPForm.styled";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faArrowDown } from "@fortawesome/free-solid-svg-icons";
+import { ethers } from "ethers";
+import toWeiSafe from "../../utils/convertToWeiSafely";
 
 import LongShort from "./LongShort";
 import Collateral from "./Collateral";
 
+const toBN = ethers.BigNumber.from;
+const scaledToWei = toBN("10").pow("18");
+
+// max uint value is 2^256 - 1
+const MAX_UINT_VAL = ethers.constants.MaxUint256;
+const INFINITE_APPROVAL_AMOUNT = MAX_UINT_VAL;
+
 interface Props {
+  address: string;
+  contractAddress: string;
   setShowSettle: React.Dispatch<React.SetStateAction<boolean>>;
+  lspContract: ethers.Contract | null;
+  erc20Contract: ethers.Contract | null;
+  web3Provider: ethers.providers.Web3Provider | null;
+  collateralBalance: ethers.BigNumber;
+  collateralPerPair: ethers.BigNumber;
+  setCollateralBalance: React.Dispatch<React.SetStateAction<ethers.BigNumber>>;
+  collateralDecimals: string;
+  longTokenBalance: ethers.BigNumber;
+  longTokenDecimals: string;
+  shortTokenBalance: ethers.BigNumber;
+  shortTokenDecimals: string;
+  refetchLongTokenBalance: () => void;
+  refetchShortTokenBalance: () => void;
 }
-const MintForm: FC<Props> = ({ setShowSettle }) => {
-  const [collateral, setCollateral] = useState("");
+
+const MintForm: FC<Props> = ({
+  lspContract,
+  erc20Contract,
+  contractAddress,
+  collateralBalance,
+  collateralPerPair,
+  address,
+  setCollateralBalance,
+  collateralDecimals,
+  longTokenBalance,
+  longTokenDecimals,
+  shortTokenBalance,
+  shortTokenDecimals,
+  refetchLongTokenBalance,
+  refetchShortTokenBalance,
+}) => {
+  const [collateral, setCollateral] = useState("uma");
   const [amount, setAmount] = useState("");
   const [longTokenAmount, setLongTokenAmount] = useState("");
   const [shortTokenAmount, setShortTokenAmount] = useState("");
 
+  const mint = useCallback(async () => {
+    if (lspContract && erc20Contract && amount) {
+      const weiAmount = toWeiSafe(amount);
```

Comment:
Is amount a perfect-precision decimal string? Is it possible there's any precision loss in amount before we get here?